apps/music/js/metadata/vorbis_picture.js

```diff
@@ -15,29 +15,27 @@ var VorbisPictureComment = (function() {
     if (kind != 3) {
       return null;
     }
-    var len = view.readUnsignedInt();
-    var mimetype = view.readASCIIText(len);
-    if (mimetype == '-->') {
-      // XXX we don't support URL (no recommended by the spec anyway)
-      console.error('URL for cover art is not supported from Vorbis comment.');
+    var mime_len = view.readUnsignedInt();
+    var mimetype = view.readLatin1Text(mime_len);
+    if (mimetype === '-->') {
+      // XXX: We don't support URLs (the spec recommends against them anyway).
+      console.error('URL for cover art is not supported.');
       return null;
     }
 
-    // We ignore these next two fields
-    len = view.readUnsignedInt();
-    view.readNullTerminatedUTF8Text(len); // desc
-    // skip w, h, bpp and indexed colours.
-    view.advance(16);
+    // Skip desc, w, h, bpp and indexed colours.
+    var desc_len = view.readUnsignedInt();
+    view.advance(desc_len + 16);
```

Comment:
This is a potential change in behavior. Is the string actually null-terminated? If so, the old code does not necessarily advance by desc_len bytes.  I don't know what the spec says, but it looks to me like there might have been a bug before, and it is gone now.  But you might double-check this.